ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -1097,21 +1097,23 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
     }
     if (!skipPartitionAndDiskAvailableCheck) {
       // 2. ensure the disk for the partition/replica is available
-      List<? extends ReplicaId> replicaIds = partition.getReplicaIds();
-      for (ReplicaId replica : replicaIds) {
-        if (replica.getDataNodeId().getHostname().equals(currentNode.getHostname())
-            && replica.getDataNodeId().getPort() == currentNode.getPort()) {
-          if (replica.getDiskId().getState() == HardwareState.UNAVAILABLE) {
-            metrics.diskUnavailableError.inc();
-            return ServerErrorCode.Disk_Unavailable;
-          }
-        }
+      ReplicaId localReplica = storageManager.getLocalReplica(partition);
```

Comment:
seems like this should be a method in the `ClusterParticipant` not `StorageManager`

Reply:
You can even turn `partitionsInCurrentNode` into a map of `PartitionId` to `ReplicaId` (`StorageManager` is certainly not the right place)

Reply:
The reason I don't use `partitionsInCurrentNode` is:   (1) it is `unmodifiableSet` and initialized in constructor. (2) if "move" replica is introduced, the new clustermap (new added partition) will be sent `storageManager` and `replicationManager` by `clusterManager`. Hence, in my opinion, `storageManager` keeps the up-to-date partitions in this node. What do you think?

Reply:
`partitionsInCurrentNode` is used in other places so it can no longer be an `unmodifiableSet` once "move replica" is introduced anyway. I don't think `StorageManager` is the right place at all.